overlord/ifacestate/handlers.go

```diff
@@ -22,33 +22,90 @@ package ifacestate
 import (
 	"errors"
 	"fmt"
+	"path"
 	"reflect"
 	"sort"
 	"strings"
 	"time"
 
 	"gopkg.in/tomb.v2"
 
+	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/i18n"
 	"github.com/snapcore/snapd/interfaces"
 	"github.com/snapcore/snapd/interfaces/hotplug"
 	"github.com/snapcore/snapd/logger"
 	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/servicestate"
 	"github.com/snapcore/snapd/overlord/snapstate"
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/quota"
 	"github.com/snapcore/snapd/timings"
 )
 
 var snapstateFinishRestart = snapstate.FinishRestart
 
+func getQuotaGroup(st *state.State, snapName string) *quota.Group {
```

Comment:
we have code like this already, we should reuse snapstate.SnapServiceOptions (which is set from servicestate.SnapServiceOptions)

Reply:
Ah this was unknown to me @pedronis - thanks for pointing that out, I have updated the code here to use snapstate instead for retrieving the quota groups.